va/va.go

```diff
@@ -400,25 +423,45 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 			reqPort = va.httpPort
 		}
 
-		dialer, err := va.resolveAndConstructDialer(ctx, reqHost, reqPort)
-		dialer.record.URL = req.URL.String()
-		// A subsequent dialing from a redirect means adding another validation
-		// record
-		validationRecords = append(validationRecords, dialer.record)
-		if err != nil {
-			return err
+		// Record validation record for previous request
```

Comment:
This comment could use more explanation of "why" we do this, as opposed to storing the validation record after a given fetch.

Also, I'd recommend "Store validation record" instead of "Record validation record" to reduce cognitive dissonance :-)